src/main.rs

```diff
@@ -11,9 +13,39 @@ use gelf::tcp_acceptor;
 use gelf::udp_acceptor;
 use gelf::unpacking::UnPackActor;
 
-fn main() {
-    let dsn = env::var("SENTRY_DSN")
-        .unwrap_or_else(|_e| panic!("You must to pass SENTRY_DSN variable from env"));
+fn get_sentry_dsn() -> Result<String, Error> {
```

Comment:
Might be easier to read with if else because you are discarding the errors. I rephrased the failed to read file since I think it might occur due to maybe e.g. permission problems or something.
```rust
fn get_sentry_dsn() -> Result<String, Error> {
    if let Ok(dsn) = env::var("SENTRY_DSN") {
        // SENTRY_DSN ENV var takes precedence over SENTRY_DSN_FILE
        return Ok(dsn);
    } else {
        if let Ok(dsn_file) = env::var("SENTRY_DSN_FILE") {
            fs::read_to_string(dsn_file).map_err(|err| {
                Error::new(
                    ErrorKind::InvalidInput,
                    format!("Failed to read SENTRY_DSN_FILE: {}", err),
                )
            })
        } else {
            Err(Error::new(
                ErrorKind::InvalidInput,
                "SENTRY_DSN and SENTRY_DSN_FILE not defined you must pass one of both as env var",
            ))
        }
    }
}
```